### PR TITLE
Use kong 2.

### PR DESCRIPTION
**Pull request by mirr254**
### What does this PR do?
Update to the latest upstream code.

### Description of tasks done.
- Use kong 2.0.1
- Use multi-stage build in Dockerfile
- Use https scheme in git submodules

### Screenshots
Screenshot 2020-02-18 at 12 02 27
Screenshot 2020-02-18 at 11 58 54

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,23 +1,21 @@
...
 # ============
 
+
 ENV GLUU_VERSION=version_4.1 \
     GG_DEPS=gluu-gateway-node-deps
 
-RUN git clone --single-branch --branch ${GLUU_VERSION} https://github.com/GluuFederation/gluu-gateway.git /tmp/${GG_DEPS} 
-
-COPY gitmodules /tmp/${GG_DEPS}
+RUN git clone --recursive --depth 1 --branch ${GLUU_VERSION} https://github.com/GluuFederation/gluu-gateway.git /tmp/${GG_DEPS}
```
Comment (altexy) [inline Dockerfile]: IMO Docker multi-stage build is the better approach.

We may do git clone ... in intermediate image, them copy all required files to the final image.

Comment (mirr254) [reply Dockerfile]: Good thought.

File: Dockerfile [hunk 2]
```diff
@@ -56,4 +52,6 @@ ENV KONG_PROXY_ACCESS_LOG=/dev/stdout \
 # Cleanup
 # =======
 
-RUN apk del build-deps
+RUN apk del build-deps 
+# restore
+USER kong
```
[the remark this answers is not in the document]
Comment (mirr254) [reply Dockerfile]: Noted

**Comment by mirr254** [comment]
Closing in favor of #9

**Pull request closed without merging**